src/main/java/org/opentripplanner/updater/vehicle_positions/PollingVehiclePositionUpdater.java

```diff
@@ -0,0 +1,93 @@
+package org.opentripplanner.updater.vehicle_positions;
+
+import com.google.transit.realtime.GtfsRealtime.VehiclePosition;
+import java.util.List;
+import org.opentripplanner.routing.graph.Graph;
+import org.opentripplanner.updater.PollingGraphUpdater;
+import org.opentripplanner.updater.WriteToGraphCallback;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Add vehicle positions to OTP patterns via a realtime source *
+ * <pre>
+ * rt.type = vehicle-positions
+ * rt.frequencySec = 60
+ * rt.sourceType = gtfs-http
+ * rt.url = http://host.tld/path
+ * rt.feedId = TA
+ * </pre>
+ */
+public class PollingVehiclePositionUpdater extends PollingGraphUpdater {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PollingVehiclePositionUpdater.class);
+
+    /**
+     * Parent update manager. Is used to execute graph writer runnables.
+     */
+    private WriteToGraphCallback saveResultOnGraph;
+
+    /**
+     * Update streamer
+     */
+    private final VehiclePositionSource vehiclePositionSource;
+    private VehiclePositionPatternMatcher vehiclePositionPatternMatcher;
+
+    private final String feedId;
+
+    @Override
+    public void setGraphUpdaterManager(WriteToGraphCallback saveResultOnGraph) {
+        this.saveResultOnGraph = saveResultOnGraph;
+    }
+
+    public PollingVehiclePositionUpdater(VehiclePositionsUpdaterParameters params) {
+        super(params);
+        var p = (VehiclePositionsUpdaterParameters) params;
+        vehiclePositionSource = new GtfsRealtimeHttpVehiclePositionSource(p.url);
+
+        LOG.info(
+                "Creating vehicle position updater running every {} seconds : {}",
+                pollingPeriodSeconds, vehiclePositionSource
+        );
+        feedId = params.feedId();
+    }
+
+    @Override
+    public void setup(Graph graph) {
+        var index = graph.index;
+        vehiclePositionPatternMatcher =
+                new VehiclePositionPatternMatcher(
+                        feedId,
+                        index::getTripForId,
+                        index::getPatternForTrip,
```

Comment:
This will not work for patterns created by realtime updaters

Reply:
You can check at https://github.com/opentripplanner/OpenTripPlanner/blob/d0e658791a8ebf11584f9e38400fe06868b86722/src/main/java/org/opentripplanner/routing/stoptimes/StopTimesHelper.java#L291-L300

Reply:
The `getLastAddedTripPattern` returns the updated pattern

Reply:
I've taken the snapshot into account in https://github.com/opentripplanner/OpenTripPlanner/pull/3981/commits/665657612e5cd5d7ce1d07d6ff2fd15ba9675eb2 but I'm not sure that the updater is the correct place to put this code. Do you have a better idea? I'm hesitant to put it on the `Graph`.

Reply:
I think this is ok as it is, we can move around this at a later stage